### ISSUE TITLE
wxStaticBoxSizer item is not rendering correctly

### ISSUE THREAD

**Issue by computergeek1507** [wxWidgets/wxWidgets#23182]
Describe the bug:
The wxStaticBoxSizer item is not rendering correctly in git master. Items in the sizer will not render or "popin" when hovered overed. The items click events work if you can guess the location. Might be caused by c5f613c9413ff11c123d8634c15dc9949fd828a9

Expected vs observed behaviour:
The items in the wxStaticBoxSizer would render and be visible.

Patch or snippet allowing to reproduce the problem:
`	StaticBoxSizer1 = new wxStaticBoxSizer(wxHORIZONTAL, this, _("Backup Directory"));
	CheckBox_Backup = new wxCheckBox(this, ID_CHECKBOX4, _("Use Show Folder"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_CHECKBOX4"));
	CheckBox_Backup->SetValue(false);
	StaticBoxSizer1->Add(CheckBox_Backup, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);`

To Reproduce:
Steps to reproduce the behaviour, please make them as detailed as possible.
For example:

1. Open Dialog with wxStaticBoxSizer
2. See items not rendering

![static_missing](https://user-images.githubusercontent.com/2293065/215279367-3d39dde6-254a-4758-a2d8-aa559242ac43.gif)

![static_popin](https://user-images.githubusercontent.com/2293065/215279256-19acfece-6d77-4338-bb5f-6842ac5d5763.gif)

### Platform and version information

- wxWidgets version  you use: https://github.com/wxWidgets/wxWidgets/commits/master 67ad244a94f9f70f9ba20aa658503aebb6064868
- wxWidgets port  you use: wxMSW
- OS  and its version: Windows 10 21H2

**Comment by PBfordev** [comment]
I can reproduce, but only when
1. The static box sizer is in a notebook.
2. The static box sizer items are not created as recommended (i.e., not as children of the `wxStaticBoxSizer::GetStaticBox()`).

```diff
diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 501caf9096..0fc06985b4 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -26,6 +26,8 @@
     #include "wx/wx.h"
 #endif
 
+#include <wx/notebook.h>
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -188,16 +190,30 @@ void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
-    wxMessageBox(wxString::Format
-                 (
-                    "Welcome to %s!\n"
-                    "\n"
-                    "This is the minimal wxWidgets sample\n"
-                    "running under %s.",
-                    wxVERSION_STRING,
-                    wxGetOsDescription()
-                 ),
-                 "About wxWidgets minimal sample",
-                 wxOK | wxICON_INFORMATION,
-                 this);
+    wxDialog dlg(this, wxID_ANY, "Test wxStaticBoxSizer");
+    wxNotebook* notebook = new wxNotebook(&dlg, wxID_ANY);
+
+    auto AddPage = [notebook](const wxString& title)
+    {
+        wxPanel* panel = new wxPanel(notebook);
+        wxBoxSizer* panelSizer = new wxBoxSizer(wxVERTICAL);
+        wxStaticBoxSizer* statBoxSizer = new wxStaticBoxSizer(wxHORIZONTAL, panel, "Backup Directory " + title);
+
+#if 1 // change 1 to 0 to make the checkboxes visible
+        statBoxSizer->Add(new wxCheckBox(panel, wxID_ANY, "checkbox 1"), 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+        statBoxSizer->Add(new wxCheckBox(panel, wxID_ANY, "checkbox 2"), 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+#else
+        statBoxSizer->Add(new wxCheckBox(statBoxSizer->GetStaticBox(), wxID_ANY, "checkbox 1"), 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+        statBoxSizer->Add(new wxCheckBox(statBoxSizer->GetStaticBox(), wxID_ANY, "checkbox 2"), 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+#endif
+        panelSizer->Add(statBoxSizer);
+
+        panelSizer->Add(new wxButton(panel, wxID_ANY, "button"));
+        panel->SetSizer(panelSizer);
+        notebook->AddPage(panel, title);
+    };
+
+    AddPage("Page 1");
+    AddPage("Page 2");
+    dlg.ShowModal();
 }
```


It seems to be related to composition as the issue disappeared when I called `panel->MSWDisableComposited();`.

**Comment by vadz** [comment]
This should be fixed by the PR linked above in an obvious (and ugly) way by just disabling compositing in this case, please check if you still have any problems with it.

Of course, this fix will result in significant flicker if you have any windows that would have profited from compositing (a.k.a. double buffering) in the same dialog, but I don't see any way to make this work with compositing being enabled.

**Comment by taler21** [comment]
Unfortunately, it does not fix the issue reported in #23239.
